src/exchanges/populate.ts

```diff
@@ -0,0 +1,331 @@
+import { pipe, tap, map } from 'wonka';
+import { Exchange, Operation } from '../types';
+import {
+  DocumentNode,
+  buildClientSchema,
+  visitWithTypeInfo,
+  TypeInfo,
+  FragmentDefinitionNode,
+  SelectionSetNode,
+  GraphQLSchema,
+  IntrospectionQuery,
+  FragmentSpreadNode,
+  GraphQLObjectType,
+  GraphQLInterfaceType,
+  ASTNode,
+} from 'graphql';
+import { visit } from 'graphql';
+
+interface ExchangeArgs {
+  schema: IntrospectionQuery;
+}
+
+/** An exchange for auto-populating mutations with a required response body. */
+export const populateExchange = ({
+  schema: ogSchema,
+}: ExchangeArgs): Exchange => ({ forward }) => {
+  const schema = buildClientSchema(ogSchema);
+  /** List of operation keys that have already been parsed. */
+  const parsedOperations = new Set<number>();
+  /** List of operation keys that have not been torn down. */
+  const activeOperations = new Set<number>();
+  /** Collection of fragments used by the user. */
+  let userFragments: UserFragmentMap = {};
+  /** Collection of type fragments. */
+  let typeFragments: TypeFragmentMap = {};
+
+  /** Handle mutation and inject selections + fragments. */
+  const handleIncomingMutation = (op: Operation) => {
+    if (op.operationName !== 'mutation') {
+      return op;
+    }
+
+    const activeSelections = Object.entries(typeFragments).reduce(
```

Comment:
If we use `Object.entries` it'll need to be polyfilled, but that is probably not really desirable since we don't need to take on a polyfill for this. (This should be auto-poyfilled, but either way we can make due without)

So let's either maybe just use good ol' `Object.keys`. Also nit: The spreading isn't necessary in a reduce. Of course it reads more nicely but we're cleaning up Graphcache to avoid unnecessary allocations to alleviate GC pressure, so it'd be awesome to carry that over to new code in `urql` as well :)